include/model.php

```diff
@@ -154,7 +169,11 @@ public function get_languages_list( $args = array() ) {
 			 * @param PLL_Language[] $languages The list of language objects.
 			 */
 			$languages = apply_filters( 'pll_after_languages_cache', $languages );
-			$this->cache->set( 'languages', $languages );
+
+			if ( $this->are_languages_ready() ) {
+				$this->cache->set( 'languages', $languages );
+			}
```

Comment:
```suggestion
			if ( $this->are_languages_ready() ) {
				$this->cache->set( 'languages', $languages );
			} else {
				_doing_it_wrong(
					__METHOD__ . '()',
					"It must not be called before the hook 'pll_pre_init'.",
					'3.4'
				);
			}
```

Reply:
As we discussed, we keep it here for a better readability.